tests/pytests/test_certifier_framework.py

```diff
@@ -0,0 +1,387 @@
+    assert pstore.delete_entry(nentries) is False
+
+    del_idx = pstore.find_entry(tag2, type2)
+    result = pstore.delete_entry(del_idx)
+    assert result is True
+
+    nentries -= 1
+
+    assert pstore.get_num_entries() == nentries
+
+# ##############################################################################
+def test_policy_store_serialize():
+    """
+    Exercise the Serialize() interface of policy_store.
+    """
+    pstore = cfm.policy_store()
+
+    tag1   = 'tag-1'
+    type1  = 'string'
+    value1 = 'Entry-1'
+    result = pstore.update_or_insert(tag1, type1, value1)
+    assert result is True
+
+    # Insert another entry, so we can check that update()'s getting the right one
+    tag2   = 'tag-2'
+    type2  = 'string'
+    value2 = 'Entry-2'
+    result = pstore.update_or_insert(tag2, type2, value2)
+    assert result is True
+
+    assert pstore.get_num_entries() == 2
+
+    (result, serialized) = pstore.Serialize()
+    assert result is True
+    assert len(serialized) > 0
+    # Does not quite work due to some embedded non-printable chars in stream
+    # print("Serialized Policy Store contents: ,", serialized, "'")
+
+# ##############################################################################
+# Test cases for class cc_trust_data()
+# ##############################################################################
+
+# ##############################################################################
+def test_cc_trust_data():
+    """ Basic exerciser of methods for an empty cc_trust_data() object."""
+
+    # fault_out = open(os.path.join(THONNY_USER_DIR, "frontend_faults.log"), mode="w")
+    # fault_out = open('/tmp/debug.out', mode="w")
+    # faulthandler.enable(fault_out)
+    cctd = cfm.cc_trust_data()
+    assert cctd.cc_all_initialized() is False
+
+    # Should fail with garbage key-algorithm names
+    public_key_alg = "public-key-alg"
+    symmetric_key_alg = "symmetric-key-alg"
+    assert cctd.cold_init(public_key_alg, symmetric_key_alg,
+                          b'fake-asn1_certificate', # passed as byte-stream
+                          "Home-domain-name", "home-host-name",
+                          8121, "service-host", 8123) is False
+
+    asn1_cert = 'some-asn1-certificate-junk-test-string'.encode()
+    assert cctd.init_policy_key(asn1_cert) is False
+
+# ##############################################################################
+def test_cc_trust_data_simulated_enclave():
+    """
+    Basic exerciser of methods for an cc_trust_data() object for a simulated
+    enclave. Go through bootstrapping interfaces, using a pre-generated policy
+    certificate from a file.
+    """
+    cctd = cfm.cc_trust_data('simulated-enclave', 'authentication',
+                             CertPyTestsDir + '/data/policy_store')
+    assert cctd.cc_all_initialized() is False
+
+    # Open the Certificate binary file for reading
+    with open(CertPyTestsDir + '/data/policy_cert_file.bin', 'rb') as cert_file:
```

Comment:
We now check-in a generated `policy_cert_file.bin`, that has been manually created using tools for simulated-enclave.

This allows us to write pytests to verify interfaces.